### DISCUSSION TITLE
Where are files like diffusion_pytorch_model.safetensors from huggingface hub saved on my disk?

### DISCUSSION THREAD

**Discussion by ali0une** [lllyasviel/stable-diffusion-webui-forge#463] [Q&A]
Hi there!

Surely a dumb question but i just can't figure it out.
Can't find related answers and tried several find commands but nothing pop up.

This morning i was playing with ControlNet Marigold Depth Preprocessor.
Forge started to download Go of files i believe from huggingface.

```
unet/config.json: 100%|██████████████████████████████████████████████████████████████████████████| 888/888 [00:00<00:00, 2.17MB/s]
diffusion_pytorch_model.safetensors: 100%|███████████████████████████████████████████████████| 3.46G/3.46G [21:41<00:00, 2.66MB/s]
vae/config.json: 100%|███████████████████████████████████████████████████████████████████████████| 611/611 [00:00<00:00, 1.85MB/s]
diffusion_pytorch_model.safetensors: 100%|█████████████████████████████████████████████████████| 335M/335M [02:05<00:00, 2.67MB/s]
scheduler/scheduler_config.json: 100%|███████████████████████████████████████████████████████████| 345/345 [00:00<00:00, 1.25MB/s]
tokenizer/vocab.json: 100%|██████████████████████████████████████████████████████████████████| 1.06M/1.06M [00:00<00:00, 1.77MB/s]
tokenizer/merges.txt: 100%|████████████████████████████████████████████████████████████████████| 525k/525k [00:00<00:00, 1.51MB/s]
tokenizer/special_tokens_map.json: 100%|██████████████████████████████████████████████████████████| 460/460 [00:00<00:00, 922kB/s]
tokenizer/tokenizer_config.json: 100%|███████████████████████████████████████████████████████████| 824/824 [00:00<00:00, 3.82MB/s]
text_encoder/config.json: 100%|██████████████████████████████████████████████████████████████████| 633/633 [00:00<00:00, 1.63MB/s]
model.safetensors: 100%|█████████████████████████████████████████████████████████████████████| 1.36G/1.36G [08:32<00:00, 2.66MB/s]
Attention optimization applied to DiffusersModelPatcher
To load target model MarigoldPipeline
```

But did not see anything related in my .cache directory.
Tried with find on my whole system ... same thing!

Could someone please help me find where are these files stored?

Thanks!

**Comment by freecoderwaifu** [comment]
Use Everything to easily find files.

https://www.voidtools.com/

Models like those are usually stored under `Users\YourName\.cache\huggingface\hub\` and respective subfolders. In my particular case, I have it further down under `\models--stabilityai--stable-cascade\snapshots\f2a84281d6f8db3c757195dd0c9a38dbdea90bb4\vqgan\diffusion_pytorch_model.safetensors`, but I also have 2 other copies. Since different extensions, UIs, and what not use different paths, it's bound to leave duplicates that have to be in their respective folders. 

You might be able to change location with `TRANSFORMERS_CACHE` or `--data-dir ` but I'm not sure as I've never tried it. 

https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Command-Line-Arguments-and-Settings

**Reply by aleph23** [reply]
Yeah, I have tons of symlinks trying to deal with model madness.

You'll also find some in ~\\.cache\torch and ~\\.insightface

**Comment by ali0une** [comment]
Hi there!

OK just found the answer.
i was using stable-diffusion-webui-forge and the huggingface model ended in the /whatever/stable-diffusion-webui/models/diffusers/models--Bingxin--Marigold directory.
